test/OmenAgentResultMapping.t.sol

```diff
@@ -0,0 +1,96 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.13;
+
+import {Test} from "forge-std/Test.sol";
+import {OmenAgentResultMapping} from "../src/OmenAgentResultMapping.sol";
+import {Prediction} from "../src/structs.sol";
+
+contract OmenAgentResultMappingTest is Test {
+    event PredictionAdded(
+        address indexed marketAddress,
+        uint16 estimatedProbabilityBps,
+        address indexed publisherAddress,
+        bytes32 txHash,
+        bytes32 ipfsHash
+    );
+
+    OmenAgentResultMapping public omenAgentResultMapping;
+    address publisher;
+    address marketAddress;
+
+    function setUp() public {
+        omenAgentResultMapping = new OmenAgentResultMapping();
+        publisher = vm.addr(1);
+        // We mock the market address value as being the address of this test contract.
+        marketAddress = address(this);
+    }
+
+    function buildPrediction(string memory input) internal view returns (Prediction memory) {
+        // Convert the input string to a bytes32 IPFS hash
+        bytes32 ipfsHash = keccak256(abi.encodePacked(input));
+        bytes32 dummyTxHash = keccak256(abi.encodePacked("dummy transaction hash"));
+        uint16 estimatedProbabilityBps = 6556; //65.56%
+        Prediction memory prediction = Prediction(publisher, ipfsHash, dummyTxHash, estimatedProbabilityBps);
+        return prediction;
+    }
+
+    function addMockPrediction(string memory input) internal returns (bytes32) {
+        Prediction memory prediction = buildPrediction(input);
+        vm.prank(publisher);
+        omenAgentResultMapping.addPrediction(marketAddress, prediction);
+        return prediction.ipfsHash;
+    }
+
+    function testAddAndGetPredictions() public {
+        // Add a mock prediction and capture the IPFS hash
+        bytes32 expectedHash1 = addMockPrediction("test-input1");
+        bytes32 expectedHash2 = addMockPrediction("test-input2");
+
+        Prediction[] memory predictions = omenAgentResultMapping.getPredictions(address(this));
+        assertEq(predictions[0].ipfsHash, expectedHash1);
+        assertEq(predictions[1].ipfsHash, expectedHash2);
+    }
+
+    function testAddPredictionRevertsForNonPublisher() public {
+        Prediction memory prediction = buildPrediction("test-input1");
+
+        vm.startPrank(vm.addr(2));
+        vm.expectRevert();
+        omenAgentResultMapping.addPrediction(marketAddress, prediction);
+        vm.stopPrank();
+    }
+
+    function testAddPredictionEmitsEvent() public {
+        Prediction memory prediction = buildPrediction("test-input1");
+
+        vm.expectEmit(true, true, false, true);
+        vm.startPrank(publisher);
+        emit PredictionAdded(
```

Comment:
Why it's need to re-define PredictionAdded event and emit it here? Does it test anything?

Reply:
That's how Foundry tests for emitted events - don't like this re-definition but it's described here 
https://book.getfoundry.sh/cheatcodes/expect-emit#examples

Reply:
Thanks, I understand the second part now, but still looks like you don't need to re-define the event? This works fine for me locally:

![Screenshot by Dropbox Capture](https://github.com/user-attachments/assets/5f407ac1-e58c-486e-94ca-091b26409b35)

Reply:
Makes sense. Added